src/v/kafka/server/quota_manager.cc

```diff
@@ -480,4 +536,43 @@ ss::future<> quota_manager::do_gc(clock::time_point expire_threshold) {
       });
 }
 
+void client_quotas_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    auto metric_defs = std::vector<ss::metrics::impl::metric_definition_impl>{};
+    auto rule_label = ss::metrics::label("quota_rule");
+    auto quota_type_label = ss::metrics::label("quota_type");
+
+    for (auto quota_type : all_client_quota_types) {
+        for (auto rule : all_client_quota_rules) {
+            metric_defs.emplace_back(sm::make_histogram(
+              "client_quota_throttle_time",
+              [this, rule, quota_type] {
+                  return get_throttle_time(rule, quota_type);
+              },
+              sm::description("Client quota throttling delay per rule and "
+                              "quota type (in micros)"),
```

Comment:
Why microseconds when the throttling actually done in milliseconds?

Usually it's preferred to use SI units (seconds), but that's only regularly applied on the `public_metrics`, although I see no reason not to prefer to follow the recomendation for new metrics.

Reply:
Fair, I've now made the metric to be reported in seconds and rounded to milliseconds (like the Kafka response).